File: Olivia/1013/house robber.java
public class Solution {
    public int rob(int[] nums) {
       int n =nums.length; 
       int []res=new int[nums.length];
       int ans=0;
       if(n==0)
       return 0;
       if(n>=1)
       res[0]=nums[0];
       if(n>=2)
        res[1]=Math.max(nums[0],nums[1]);
        if(n>=3)
        res[2]=Math.max(nums[0]+nums[2],nums[1]);
        if(n>2){
            for(int i=3;i<n;i++){
                res[i]=Math.max(res[i-3],res[i-2])+nums[i];
            }
        }
        for(int i=0;i<n;i++){
            ans =Math.max(ans,res[i]);
        }
        return ans;
    }
}

File: Sen/House_Robber.py
class Solution(object):
    # def rob(self, nums):
    #     """
    #     :type nums: List[int]
    #     :rtype: int
    #     """
    #     n = len(nums)
    #     dp = [0] * (n + 1)
    #     if n:
    #     	dp[1] = nums[0]
    #     for i in range(2, n + 1):
    #     	dp[i] = max(dp[i - 1], dp[i - 2] + nums[i - 1])
    #     return dp[n]

    def rob(self, nums):
    	n = len(nums)
    	odd, even = 0, 0
    	for i in range(n):
    		if i % 2:
    			odd = max(odd + nums[i], even)
    		else:
    			even = max(even + nums[i], odd)
    	return max(odd, even)   

nums = [4, 8, 2, 2, 10]
i = Solution()
print i.rob(nums)